railgun/pkg/config/config.go

```diff
@@ -117,6 +118,11 @@ func (c *Config) FlagSet() *pflag.FlagSet {
 			"Define the rate (in seconds) in which configuration updates will be applied to the Kong Admin API. (default: %g seconds)",
 			proxy.DefaultSyncSeconds,
 		))
+	flagSet.Float32Var(&c.ProxyTimeoutSeconds, "timeout-seconds", proxy.DefaultSyncSeconds,
+		fmt.Sprintf(
+			"Define the rate (in seconds) in which the timeout configuration will be applied to the Kong client. (default: %g seconds)",
+			proxy.DefaultSyncSeconds,
+		))
 	flagSet.StringVar(&c.KongCustomEntitiesSecret, "kong-custom-entities-secret", "", `A Secret containing custom entities for DB-less mode, in "namespace/name" format`)
 
 	// Kubernetes configurations
```

Comment:
As per the PR checklist, we should add an item to the KIC 2.0 `CHANGELOG.md` noting the inclusion of this new flag.